luigi/tools/range.py

```diff
@@ -516,7 +516,7 @@ def missing_datetimes(self, task_cls, finite_datetimes):
         except NotImplementedError:
             return infer_bulk_complete_from_fs(
                 finite_datetimes,
-                lambda d: task_cls(self.datetime_to_parameter(d)),
```

Comment:
this change removes the use of `task_cls` as an argument to `missing_datetimes()`.  So it seems to me that either this patch should remove that argument (but I'm not sure what side effects that might have... when is `self.of_cls` ever different from `task_cls`?), or that instead of dropping `task_cls` as an argument to `_instantiate_task_cls()` this should call `self._instantiate_task_cls(task_cls, self.datetime_to_parameter(d))` instead.

In other words, we either need `task_cls` or we don't, and if we don't we shouldn't pass it around and should just use `self.of_cls` instead.